nebula_hw_interfaces/include/nebula_hw_interfaces/nebula_hw_interfaces_hesai/hesai_cmd_response.hpp

```diff
@@ -251,157 +254,421 @@ struct HesaiInventory
+  {
+    switch (value.ptp_status) {
+      case 0:
+        return "free run";
+      case 1:
+        return "tracking";
+      case 2:
+        return "locked";
+      case 3:
+        return "frozen";
+      default:
+        return "Unknown";
+    }
+  }
+};
 
-    os << ", ";
-    os << "gps_pps_lock: " << static_cast<int>(arg.gps_pps_lock);
-    os << ", ";
-    os << "gps_gprmc_status: " << static_cast<int>(arg.gps_gprmc_status);
-    os << ", ";
-    os << "startup_times: " << arg.startup_times;
-    os << ", ";
-    os << "total_operation_time: " << arg.total_operation_time;
-    os << ", ";
-    os << "ptp_clock_status: " << static_cast<int>(arg.ptp_clock_status);
-    os << ", ";
-    os << "reserved: ";
+inline std::ostream & operator<<(std::ostream & os, HesaiLidarStatusAT128 const & arg)
+{
+  os << "temperature: " << arg.value.temperature << '\n';
+  os << "gps_pps_lock: " << +arg.value.gps_pps_lock << '\n';
+  os << "gps_gprmc_status: " << +arg.value.gps_gprmc_status << '\n';
+  os << "startup_times: " << arg.value.startup_times << '\n';
+  os << "total_operation_time: " << arg.value.total_operation_time << '\n';
+  os << "ptp_status: " << +arg.value.ptp_status << '\n';
+  for (size_t i = 0; i < sizeof(arg.value.reserve); i++) {
+    if (i != 0) {
+      os << ' ';
+    }
+    os << std::hex << std::setfill('0') << std::setw(2) << (+arg.value.reserve[i]);
+  }  
+  return os;
+}
 
-    for (size_t i = 0; i < sizeof(arg.reserved); i++) {
-      if (i != 0) {
-        os << ' ';
-      }
-      os << std::hex << std::setfill('0') << std::setw(2) << (static_cast<int>(arg.reserved[i]));
+struct HesaiLidarStatusOT128: public HesaiLidarStatusBase
+{
+  struct Internal: public HesaiLidarStatusBase::Internal{
+    big_int32_buf_t temperature[8];
+    uint8_t gps_pps_lock;
+    uint8_t gps_gprmc_status;
+    big_int32_buf_t startup_times;
+    big_int32_buf_t total_operation_time;
+    uint8_t ptp_status;
+    big_uint32_buf_t humidity;
+    unsigned char reserve[1];
+  };
+  HesaiLidarStatusOT128(Internal value) : value(value) {}
+  [[nodiscard]] const HesaiLidarStatusBase::Internal & get() const override
+  {
+    return value;
+  }
+  Internal value;
+  std::string get_str_gps_pps_lock()
+  {
+    switch (value.gps_pps_lock) {
+      case 1:
+        return "Lock";
+      case 0:
+        return "Unlock";
+      default:
+        return "Unknown";
     }
-    os.copyfmt(initial_format);
+  }
+  std::string get_str_gps_gprmc_status()
+  {
+    switch (value.gps_gprmc_status) {
+      case 1:
+        return "Lock";
+      case 0:
+        return "Unlock";
+      default:
+        return "Unknown";
+    }
+  }
+  std::string get_str_ptp_clock_status()
+  {
+    switch (value.ptp_status) {
+      case 0:
+        return "free run";
+      case 1:
+        return "tracking";
+      case 2:
+        return "locked";
+      case 3:
+        return "frozen";
+      default:
+        return "Unknown";
+    }
+  }
```

Comment:
These are identical to the functions in `HesaiLidarStatusAT128`. Please delete them here, and move the ones from `HesaiLidarStatusAT128` to `HesaiLidarStatusBase`. Duplicated code is hard to maintain.